x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/components/search_index/pipelines/ml_inference/add_inference_pipeline_flyout.tsx

```diff
@@ -147,11 +149,29 @@ export const AddInferencePipelineHorizontalSteps: React.FC = () => {
     isPipelineDataValid,
   } = useValues(MLInferenceLogic);
   const { onAddInferencePipelineStepChange } = useActions(MLInferenceLogic);
+
+  /**
+   * Convenience function for determining the status of a step in the horizontal nav.
+   * @param currentStep The current step in the pipeline.
+   * @param otherStep The step to compare against.
+   * @returns The status of the step.
+   */
+  const getStepStatus = (
+    currentStep: AddInferencePipelineSteps,
+    otherStep: AddInferencePipelineSteps
+  ) => {
+    return currentStep > otherStep
+      ? 'complete'
+      : currentStep === otherStep
+      ? 'current'
+      : 'incomplete';
+  };
+
   const navSteps: EuiStepsHorizontalProps['steps'] = [
     {
       // Configure
       onClick: () => onAddInferencePipelineStepChange(AddInferencePipelineSteps.Configuration),
-      status: isConfigureStepValid ? 'complete' : 'disabled',
+      status: step > AddInferencePipelineSteps.Configuration ? 'complete' : 'current',
```

Comment:
Why are you making this change? the current code should make the step complete with the form values for this step are populated and that should be the same value that enables the next step button. We may need to introduce `isFieldsStepValid` since we split the configuration step if we never updated the `isConfigureStepValid` selector.

Reply:
@TattdCodeMonkey The validation for the Continue button is [still there](https://github.com/elastic/kibana/pull/163723/files#diff-4fdf2acafb75e6058d2c53a8602ccb4a578dcee9ff70ff506fe759184b86dce7R274), and it gets enabled with the right boolean selectors. Same goes for [clicking into](https://github.com/elastic/kibana/pull/163723/files#diff-4fdf2acafb75e6058d2c53a8602ccb4a578dcee9ff70ff506fe759184b86dce7R185) each step.

I found this UX cleaner; the step numbers turn to complete once the user has passed them. Previously the first two steps became complete when the corresponding checks were satisfied, but the last steps were never complete (only disabled -> incomplete -> current). The new UX is also more aligned with other uses of the horizontal step, see App Search -> Create a new engine.

You can see it in action in this screencap:
![updat_mappings_non_elser](https://github.com/elastic/kibana/assets/14224983/3b03fdeb-8bb9-4275-9f91-1ee49f2c5a10)

Let me know what you think.

Reply:
ok, if it's in line with what we do elsewhere I don't have any issue. I was just thinking about those status values differently when originally implementing these steps.

Reply:
So I [asked](https://elastic.slack.com/archives/C7QC1JV6F/p1692044788765119) in the EUI slack channel for guidance, and I agree with their take:

> My thinking is, hitting `next` is somewhat equal to “saving” or “submitting”. Before that, users might have completed the required step but nothing prevents from deleting their answer or make it invalid somehow. In that case you might end up switching the state of the step (incomplete | complete | incomplete and so on…) a bunch of times. Seems better to considered it `complete` once they’ve somehow committed their input (by hitting next).

Conclusion: Change the step only on submission and validation (via the next button), not on a form change.